src/mscorlib/shared/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -1056,14 +1056,14 @@ protected internal struct EventData
             /// <param name="reserved">Value for reserved: 2 for per-provider metadata, 1 for per-event metadata</param>
             internal unsafe void SetMetadata(byte* pointer, int size, int reserved)
             {
-                this.m_Ptr = (long)(ulong)(UIntPtr)pointer;
+                this.m_Ptr = (ulong)pointer;
```

Comment:
💭 The C# Interactive window doesn't allow unsafe code, but I'm expecting this to work based on the following (emphasis mine):

> Mappings between pointers and integers are implementation-defined. However, on 32\* and 64-bit CPU architectures with a linear address space, conversions of pointers to or from integral types **typically behave exactly like conversions of `uint` or `ulong` values, respectively, to or from those integral types**.